packages/matchbox/src/components/Slider/tests/Slider.test.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import Slider from '../Slider';
+import { mount } from 'enzyme';
+import { act } from 'react-dom/test-utils';
+import * as geometry from '../../../helpers/geometry';
+
+const map = {};
+global.addEventListener = jest.fn((event, cb) => {
+  map[event] = cb;
+});
+
+global.removeEventListener = jest.fn((event, cb) => {
+  delete map[event];
+});
+
+describe('Slider component', () => {
+  let onChange;
+
+  beforeEach(() => {
+    onChange = jest.fn();
+    geometry.getRectFor = jest.fn(() => ({
+      left: 0, right: 0, width: 200
+    }));
+  });
+  const subject = (props = {}) => mount(<Slider {...props}/>);
+
+  it('should render default state correctly', () => {
+    expect(subject()).toMatchSnapshot();
+  });
+
+  it('should render a default value', () => {
+    const slider = subject({ defaultValue: 50 });
+    expect(slider.find('.Track').prop('style').width).toBe(100);
```

Comment:
Since we're using full DOM rendering here, should we assert on the DOM rather than a prop we use to render to the DOM? That way, we're sure of the rendered result. Maybe a helper would make that less icky:

```js
expect(slider.find('.Track').toHaveStyle({ width: '100px' });
expect(slider.find('.Handle')).toHaveStyle({ left: '100px' });


expect.extend({
  toHaveStyle(received, expected) {
    const styleSpec = Object.keys(expected).map((prop) => `${prop}:\\s${expected[prop]};`);
    const receivedStyle = received.getDOMNode().getAttribute('style');
    return {
      pass: new RegExp(styleSpec).test(receivedStyle),
      message: () =>
        `expected ${this.utils.printReceived(
          receivedStyle
        )} to have style ${this.utils.printExpected(expected)}`
    };
  }
});
```

Reply:
Hmm yeah thats a good idea